test/Test/Toml/Gen.hs

```diff
@@ -161,8 +162,39 @@ genInteger = toInteger <$> Gen.int (Range.constantBounded @Int)
 genDouble :: MonadGen m => m Double
 genDouble = Gen.double $ Range.constant @Double (-1000000.0) 1000000.0
 
+genAlphaNum :: MonadGen m => m Text
+genAlphaNum = Gen.text (Range.constant 0 256) Gen.alphaNum
+
+-- | Generatates control sympol.
+genEscapeSequence :: MonadGen m => m Text
+genEscapeSequence = Gen.element
+    [ "\n", "\b", "\f", "\r", "\t", "\\", "\"" ]
+
+-- | Generatates n length list of hex chars.
+genDiffHex :: MonadGen m => Int -> m String
+genDiffHex n = sequenceA $ replicate n Gen.hexit
+
+-- | Generates unicode color string (u1234)
+genUniHex4Color :: MonadGen m => m Text
+genUniHex4Color = do
+    hex <- genDiffHex 4
+    pure . Text.pack $ "\\u" ++ hex
+
+-- | Generates unicode color string (u12345678)
+genUniHex8Color :: MonadGen m => m Text
+genUniHex8Color = do
+    hex <- genDiffHex 8
+    pure . Text.pack $ "\\U" ++ hex
+
 genText :: MonadGen m => m Text
-genText = Gen.text (Range.constant 0 256) Gen.alphaNum
+genText = Text.concat <$>
+    (Gen.shuffle =<< sequenceA
```

Comment:
Previously `genText` generated random `Text` between length 0 and 256. But here it looks like it's length will be always 4, because you call 4 different generators only once and then shuffle. What would be really nice is call generator for any possible character:

* alpha-numeric
* punctuation
* escape sequence
* hex4
* hex8

And then from 0 to 256 times choose one generator randomly each time. Probably you can use `Gen.choice` for this.

Reply:
Output are different and long.
```haskell
*Test.Toml.Gen> Gen.sample genText
"\n\\u15A1mU90PmF8bIIvPsgKgoO3t4PiCS97TzbJMIHme3vXxtqJiLrTL2xLNMY9PofL23Zkta7namSESbKlNW8aEdRyEONkkS8lQMaKRkhWC6W6ox0kRcHUpvh1tnVswUAqaa0Rb90D5bwMG6bIr0m2HZCcKIXza224Lw0D9kI9lonuhgJB2885HgZ5SirnLVWu6zH3NX9wptnqqxv62vbRRnbfpSfNkVNB\\U5A3390Ad"
*Test.Toml.Gen> Gen.sample genText
"3MPdznNiNoIyR8W3HHJRqnAqkmBPIbv7quo7KbOJPiYgXJ9LyzAt8axJvRxhaGYc17JXiraGIJdBadNJxySiEWbxGrSGQllCFJXF8MiCczxWK0LUNnyE6SSAMp0LPIGLTQRT3mLisOFWPxySna64aAsCxV1GlLBofBsZd\\UAf3Fcf92\f\\ucE17"
*Test.Toml.Gen> Gen.sample genText
"\\uFCbc\\U3E04906C\nfVCI5reFLt2zHcAdAD8eu557JGCdMpdWhPuG5UOg6HwNMfdmEui5ircBp5dnKK1"
*Test.Toml.Gen> Gen.sample genText
"\\\\UDEAF6dBd\\uEc2BYTnznbcWgWq5XOtbX2EY55cyHMRFLhycaXW7FkUycaivPMRJVythfwGBotGCXRYqRiPHwld3f3eHylG2IMzBPIUgMHLL9Kch0zB1UXRMDEDzeZ81lG3bCWExZfRSZmUXHvnQ"
*Test.Toml.Gen>
```
Ok?

Reply:
Okay, not exactly 4 characters. But you string always contains from 0 to 256 alphanumeric values and exactly one escape sequence and exactly one hex 4 unicode and exactly one hex 8 unicode.